pallets/dkg-metadata/src/lib.rs

```diff
@@ -737,150 +740,172 @@ impl<T: Config> Pallet<T> {
 	}
 
 	fn submit_genesis_public_key_onchain(block_number: T::BlockNumber) -> Result<(), &'static str> {
-		let mut agg_key_ref = StorageValueRef::persistent(AGGREGATED_PUBLIC_KEYS_AT_GENESIS);
-		let mut submit_at_ref = StorageValueRef::persistent(SUBMIT_GENESIS_KEYS_AT);
-		const RECENTLY_SENT: &str = "Already submitted a key in this session";
-		let submit_at = submit_at_ref.get::<T::BlockNumber>();
-
-		if let Ok(Some(submit_at)) = submit_at {
-			if block_number < submit_at {
-				frame_support::log::debug!(target: "dkg", "Offchain worker skipping public key submmission");
-				return Ok(())
+		let mut lock = StorageLock::<Time>::new(AGGREGATED_PUBLIC_KEYS_AT_GENESIS);
```

Comment:
This one is not correct, as you here using the storage key `AGGREGATED_PUBLIC_KEYS_AT_GENESIS` as the storage lock key. It should be like the following:
```suggestion
		let mut lock = StorageLock::<Time>::new(SUBMIT_SIGNED_PROPOSAL_ON_CHAIN_LOCK);
```